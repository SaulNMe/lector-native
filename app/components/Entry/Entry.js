import React, { Component } from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import { Colors } from 'horus-native/app/styles';
import SubtitleText from 'horus-native/app/components/SubtitleText';
import BodyText from 'horus-native/app/components/BodyText';


import styles from './EntryStyle';

export default class Entry extends Component {
	render () {
		const bgColor = this.props.registrationMode === 'in' ? { backgroundColor: Colors.lightBlue} : { backgroundColor: Colors.leave};
		return (
			<View style={[styles.row, styles.fullWidth]}>
				<View style={styles.timeContainer}>
					<SubtitleText 
						text={this.props.entryTime}
						weight='light'
					/>
				</View>

				<View style={[styles.marginHorizontal, styles.circleContainer]}>
					<View style={[styles.circle, bgColor]} testID='entry-color-indicator'></View>
				</View>

				<View style={styles.flex1}>
					<SubtitleText
						weight='bold' 
						text={ this.props.registrationMode === 'out' ? 'Salida' : 'Entrada'}
						testID='entry-message'
					/>
					<BodyText 
						text={this.props.name}
						weight='light'
					/>
				</View>
			</View>
		);
	}
}

Entry.propTypes = {
	entryTime: PropTypes.string,
	registrationMode: PropTypes.string,
	name: PropTypes.string

}

Entry.defaultProps = {
	entryTime: "00:00",
	registrationMode: 'out',
	name: '--'
}
