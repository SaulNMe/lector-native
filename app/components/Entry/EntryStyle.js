import { StyleSheet } from 'react-native';
import { ApplicationStyles } from 'horus-native/app/styles';

export default StyleSheet.create({
	...ApplicationStyles.screen,
	circle: {
		width:10,
		height:10,
		borderRadius:10,
		backgroundColor: '#444',
		marginRight: 5,
	},
	circleContainer: {
		height: 24,
		justifyContent: 'center',
	},
	timeContainer: {
		width: 50,
	}
});
