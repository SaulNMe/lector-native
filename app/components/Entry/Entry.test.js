import React from 'react';
import { shallow } from 'enzyme';
// import Colors from './Colors'

import Entry from './Entry';

function createTestProps (props) {
	return {
		// common props
		entryTime: "13:37",
		isExit: false,
		name: 'name',
		// allow to override common props
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<Entry {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});
	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	describe('With default props', () => {
		beforeEach(() => {
			wrapper = createWrapper();
		});
		it('has a default isExit set to false', () => {
			expect(wrapper.instance().props.isExit).toBe(false);
		});
	});
	describe('when isExit is set to false', () => {
		beforeEach(() => {
			const props = createTestProps({ isExit: false })
			wrapper = createWrapper(props);
		});
		it('says Entrada', () => {
			expect(wrapper.findWhere(node => node.prop('testID') === 'entry-message').props().text).toBe('Entrada');
		});
	});
	describe('when isExit is set to true', () => {
		beforeEach(() => {
			const props = createTestProps({ isExit: true })
			wrapper = createWrapper(props);
		});
		it('says Salida', () => {
			expect(wrapper.findWhere(node => node.prop('testID') === 'entry-message').props().text).toBe('Salida');
		});
	});
})

// describe('callbacks');
// describe('lifecycle');
