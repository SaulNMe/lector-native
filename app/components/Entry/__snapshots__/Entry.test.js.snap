// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<Component
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "width": "100%",
      },
    ]
  }
>
  <Component
    style={
      Object {
        "width": 50,
      }
    }
  >
    <SubtitleText
      text="13:37"
      weight="light"
    />
  </Component>
  <Component
    style={
      Array [
        Object {
          "marginHorizontal": 16,
        },
        Object {
          "height": 24,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "backgroundColor": "#444",
            "borderRadius": 10,
            "height": 10,
            "marginRight": 5,
            "width": 10,
          },
          Object {
            "backgroundColor": "#7DDFDA",
          },
        ]
      }
      testID="entry-color-indicator"
    />
  </Component>
  <Component
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SubtitleText
      testID="entry-message"
      text="Entrada"
      weight="bold"
    />
    <BodyText
      text="name"
      weight="light"
    />
  </Component>
</Component>
`;
